src/views/Pottery/components/FinishedRounds/AllHistoryCard/index.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState, useRef, useEffect } from 'react'
+import styled from 'styled-components'
+import { Card, Text, Skeleton, CardHeader, Flex, BunnyPlaceholderIcon } from '@pancakeswap/uikit'
+import { useTranslation } from 'contexts/Localization'
+import { useAppDispatch } from 'state'
+import { setFinishedRoundInfoFetched, fetchPotteryRoundData } from 'state/pottery/index'
+import { usePotteryData } from 'state/pottery/hook'
+import RoundSwitcher from 'views/Lottery/components/AllHistoryCard/RoundSwitcher'
+import { getDrawnDate } from 'views/Lottery/helpers'
+import PreviousRoundCardBody from './PreviousRoundCardBody'
+
+const StyledCard = styled(Card)`
+  width: 100%;
+  margin: auto;
+
+  ${({ theme }) => theme.mediaQueries.md} {
+    width: 579px;
+  }
+`
+
+const StyledCardHeader = styled(CardHeader)`
+  display: flex;
+  justify-content: space-between;
+  z-index: 2;
+  background: none;
+  border-bottom: 1px ${({ theme }) => theme.colors.cardBorder} solid;
+`
+
+const AllHistoryCard = () => {
+  const {
+    t,
+    currentLanguage: { locale },
+  } = useTranslation()
+  const dispatch = useAppDispatch()
+  const { publicData, finishedRoundInfo } = usePotteryData()
+  const currentPotteryId = publicData.lastDrawId
+
+  const [latestRoundId, setLatestRoundId] = useState(null)
+  const [selectedRoundId, setSelectedRoundId] = useState('')
+  const timer = useRef(null)
+
+  useEffect(() => {
+    if (currentPotteryId) {
+      const currentPotteryIdAsInt = currentPotteryId ? parseInt(currentPotteryId) : null
+      const mostRecentFinishedRoundId = currentPotteryIdAsInt > 0 ? currentPotteryIdAsInt - 1 : ''
+      setLatestRoundId(mostRecentFinishedRoundId)
+      setSelectedRoundId(mostRecentFinishedRoundId.toString())
+    }
+  }, [currentPotteryId])
+
+  useEffect(() => {
+    setFinishedRoundInfoFetched(true)
+
+    timer.current = setInterval(() => {
+      if (selectedRoundId) {
+        dispatch(fetchPotteryRoundData(selectedRoundId))
+      }
+      clearInterval(timer.current)
+    }, 1000)
```

Comment:
could you tell me why do we need setInterval here?

Reply:
If not, we well cause every time user type will keep fetch data.

Reply:
useDebouncedChangeHandler can fulfill the goal?

Reply:
I think so, let me take a look.

Reply:
@chef-jojo  Updated.